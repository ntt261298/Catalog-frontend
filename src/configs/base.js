export default {
  apiHost: 'http://localhost:5000',
};
