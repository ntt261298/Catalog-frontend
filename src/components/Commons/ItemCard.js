import React from 'react';
import { Button, Card } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { PropTypes } from 'prop-types';

export const ItemCard = (props) => {
  const { id, title, category_id } = props.item;
  return (
    <Card key={id} className="item-card">
      <Card.Body>
        <Card.Title style={{ borderBottom: '1px solid #f7f7f7' }}>{ title }</Card.Title>
        <Link to={`/category/${category_id}/item/${id}`}>
          <Button variant="primary">
              View Detail
          </Button>
        </Link>
      </Card.Body>
    </Card>
  );
};


ItemCard.propTypes = {
  item: PropTypes.object.isRequired,
};

export default ItemCard;
