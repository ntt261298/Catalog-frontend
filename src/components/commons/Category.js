import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Link } from 'react-router-dom';
import getCategories from '../../actions/category';

export class Category extends Component {
  componentDidMount() {
    this.props.getCategories();
  }

  render() {
    const { categories } = this.props;
    return (
      <div className="category">
        <h2>Catagories</h2>
        <ul>
          {
            categories.map(({ id, name }) => (
              <li key={id}>
                <Link to={`/category/${id}`}>{ name }</Link>
              </li>
            ))
          }
        </ul>
      </div>
    );
  }
}

Category.propTypes = {
  getCategories: PropTypes.func.isRequired,
  categories: PropTypes.array.isRequired,
};

const mapStateToProps = state => ({
  categories: state.category.categories,
});

export default connect(mapStateToProps, { getCategories })(Category);
