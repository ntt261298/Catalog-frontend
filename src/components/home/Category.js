import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import getCategories from '../../actions/category';

class Category extends Component {
  componentDidMount() {
    // eslint-disable-next-line react/destructuring-assignment
    this.props.getCategories();
  }

  render() {
    const { categories } = this.props;
    return (
      <div className="category">
        <h2>Catagories</h2>
        <ul>
          {
            categories.map(({ id, name }) => (
              <li key={id}>{ name }</li>
            ))
          }
        </ul>
      </div>
    );
  }
}

Category.propTypes = {
  getCategories: PropTypes.func.isRequired,
  categories: PropTypes.arrayOf.isRequired,
};

const mapStateToProps = state => ({
  categories: state.category.categories,
});

export default connect(mapStateToProps, { getCategories })(Category);
