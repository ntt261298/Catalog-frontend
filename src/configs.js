import {
  configure, shallow, mount, render,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// Enzyme config
configure({ adapter: new Adapter() });

// API host config
const defaultHost = 'http://localhost:5000';

export {
  shallow, mount, render, defaultHost,
};
