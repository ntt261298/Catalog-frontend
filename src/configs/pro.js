import base from 'configs/base';

export default {
  ...base,
};
