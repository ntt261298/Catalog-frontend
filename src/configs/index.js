import {
  configure, shallow, mount, render,
} from 'enzyme/build';
import Adapter from 'enzyme-adapter-react-16/build';
import dev from 'configs/dev';
import pro from 'configs/pro';

// Enzyme config
configure({ adapter: new Adapter() });

let configs;

if (process.env.REACT_APP_ENV === 'dev') {
  configs = dev;
} else if (process.env.REACT_APP_ENV === 'pro') {
  console.log('pro');
  configs = pro;
} else {
  // Default configs is dev
  configs = dev;
}

export { shallow, mount, render };

export default configs;
