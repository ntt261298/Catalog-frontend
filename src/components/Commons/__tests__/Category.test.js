import React from 'react';
import { shallow } from 'configs';
import { Category } from 'components/Commons/Category';

describe('components/Commons/Category', () => {
  let wrapper;
  let props;
  let linkCategory;

  const update = () => {
    linkCategory = wrapper.find('Link');
  };

  const setup = () => {
    wrapper = shallow(
      <Category {...props} />,
    );
    wrapper.instance().changeActive = jest.fn();
    update();
  };

  beforeEach(() => {
    props = {
      categories: [
        {
          id: 1,
          items: [],
          name: 'Category 1',
        },
        {
          id: 2,
          items: [],
          name: 'Category 2',
        },
      ],
      getCategories: jest.fn().mockImplementation(() => Promise.resolve('message')),
    };
  });

  it('should render list-categories', () => {
    setup();
    // Expect the wrapper object to be defined
    expect(wrapper).toMatchSnapshot();
  });

  it('should call getCategories in componentDidMount', () => {
    setup();
    // Expect the wrapper object to be defined
    expect(props.getCategories).toHaveBeenCalled();
  });

  it('should call changeActive when Link is clicked', () => {
    setup();
    // Expect the wrapper object to be defined
    linkCategory.at(0).simulate('click');
    expect(wrapper.instance().changeActive).toHaveBeenCalled();
  });
});
