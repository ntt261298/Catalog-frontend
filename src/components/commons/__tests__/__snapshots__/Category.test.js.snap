// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/commons/Category should render list-categories 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
